DBSCE/Overview.md

```diff
@@ -0,0 +1,219 @@
+
+This is a pre-requisite for DBSC(E) to work.
+
+Device Registration Client is a process where the user registers the device with the IdP and is expected to be a once-in-a-lifetime operation.
+
+The device registration establishes trust between the device and a service that maintains a directory of all devices. This document does not cover the protocol of device registration, but it assumes that during device registration, some asymmetric keys are shared between the client and the service, typically a device key and some other keys necessary for the secure device communication. A client software component that performs the device registration is called a _device registration client_. As mentioned above, the key assumption in DBSC(E) is that device registration happened in a clean room environment, and it is the responsibility of the device owner to ensure this.
+
+One device registration client can manage multiple devices on the same physical device. There also can be multiple device registration clients on the same device. The device registration client can be owned and supported by:
+
+- Operating system - the device gets registered when the OS is installed.
+- Browser - the device gets registered when the browser is installed.
+- Device management software (MDM provider) - the device gets registered when the MDM is enrolled.
+- Third-party software vendor - the device gets registered according to the vendor rules.
+
+DBSC(E) aims to support most of these scenarios. It does not define the device registration protocol amd is only concerned with the keys generated in a "clean room" state and the management of the generated keys to prove device binding.
+
+### Local Key Helper
+
+DBSC(E) introduces the concept of `Local Key Helper` which can be mapped to the `TPM` or any `Key generation helper` for the consumer case.
+**Local Key Helper** is an integral part of the the **Device Registration Client**, a software interface responsible for the DBSC Key management. It can be Public or Private and is expected to be either shipped as a part of a given enterprise framework (with the IdP/OS) or can be installed by a provider in compliance with the protocol expanded below.
+
+From the deployment point of view there are two types of local key helpers: _well-known_(_private_) and _third party_(_public_)
+
+- _Public local key helper_ or _third party_: Can be accessed by any Identity Provider (IdP). Typically owned by a provider different from the IdP, communicates with the IdP through a well-defined public protocol. A third party local key helper has a special deployment mechanism.
+- _Private local key helper_ or _well known_ : Can be specific to an IdP. Typically owned by the IdP and will have a private protocol to communicate with the IdP. Comes with either OS or built into the browser. It assumed that well-known key helpers are **trusted and enabled by default** in a browser. A browser knows how to activate a well-known key helper.
```

Comment:
I'm a bit confused by this categorization. I assume our local key helper for Mac will be private, but will not be built into OS or browser. Would it still be categorized as private?

Reply:
I'm still getting through the KeyGeneration doc, but so far it's unclear to me what happens if a browser were to implement their own LocalKeyHelper for a given domain, and an additional private LocalKeyHelper were registered for the same domain.  Is there an expectation it's all driven by a registry like model where any domain could be overwritten by an enterprise admin or similar actor?

Reply:
IDP provides a priority list of helpers and browser should pick first helper on that list that it finds present. In your example, it depends on which of those helpers was first on the list

Reply:
Yeah, this is the detail we want the spec to get it right - Precedence of picking a Local Key Helper, and it is for now whatever is the first in the list. 

>> I'm a bit confused by this categorization. I assume our local key helper for Mac will be private, but will not be built into OS or browser. Would it still be categorized as private?
As long as it comes with the box, and in your case if the installation is mandated. `private` or `well known` as we have envisioned is more about IDP `knows` what Local Key Helper to pick - and it could be that a certain IDP will always prefer to have some Local Key Helper as default.

Reply:
Ok, can we explain it that way then? Private or well known is a key helper that is known/trusted by the IDP?

Reply:
Added.